From 057f1612e68771bbf18806633a6f1d516ac2bd9b Mon Sep 17 00:00:00 2001
From: Leon Leijssen <leon.leijssen@topic.nl>
Date: Mon, 4 May 2020 10:17:40 +0200
Subject: [PATCH 07/12] drm: xlnx: pl_disp: fix missing xlnx_crtc_unregister()

During shutdown of the system WARN_ON(!list_empty(&helper->xlnx_crtcs)) in
xlnx_crtc.c xlnx_crtc_helper_fini() was being triggered. Fix it by
unregistering the registered crtc.

Signed-off-by: Leon Leijssen <leon.leijssen@topic.nl>
---
 drivers/gpu/drm/xlnx/xlnx_pl_disp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
index a1d720d0271a..f5f1ce64266d 100644
--- a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
+++ b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
@@ -524,6 +524,7 @@ static void xlnx_pl_disp_unbind(struct device *dev, struct device *master,
 {
 	struct xlnx_pl_disp *xlnx_pl_disp = dev_get_drvdata(dev);
 
+	xlnx_crtc_unregister(xlnx_pl_disp->drm, &xlnx_pl_disp->xlnx_crtc);
 	drm_plane_cleanup(&xlnx_pl_disp->plane);
 	drm_crtc_cleanup(&xlnx_pl_disp->xlnx_crtc.crtc);
 }
-- 
2.17.1

